src/vm/errors.rs

```diff
@@ -60,6 +60,7 @@ pub enum InterpreterError {
     FailureConstructingTupleWithType,
     FailureConstructingListWithType,
     InsufficientBalance,
+    InvalidPoisonMicroblockTransaction(String),
```

Comment:
Related to my suggestion to move the poison microblock evaluation out of the `vm::functions` module, it seems like this shouldn't be a possible interpreter error.

Reply:
Sure, but because `handle_poison_microblock()` is needs to return `InterpreterResult` (a consequence of having to run within `OwnedEnvironment::execute_in_env()`), what error type would you recommend?  I could either introduce this error type, or add other more-specific error types that will only get used in this code-path, or try and shoe-horn errors from `vm::errors` into the relevant error paths.  In particular, I need error types to describe somewhat-specific errors:

* Invalid signature validation (for when the poison-microblock payload is corrupt)
* The public key hash hasn't been seen in this fork
* The reward that this poison-microblock transaction is targeting has matured

Reply:
I could just use a catch-all `InterpreterError::InterpreterError(String)` if you prefer.

Reply:
I'd prefer updating the signature of `execute_in_env` to like:

```rust
    pub fn execute_in_env<F, A, E>(
        &mut self,
        sender: Value,
        f: F,
    ) -> std::result::Result<(A, AssetMap, Vec<StacksTransactionEvent>), E>
    where
        F: FnOnce(&mut Environment) -> std::result::Result<A, E>,
```

That way, `handle_poison_microblock` can return a `chainstate::stacks::Error` (or any other error type) without forcing it to be a `InterpreterError`.

Reply:
Addressed in ee12847